x-pack/plugins/observability_solution/observability/public/pages/alerts/components/alert_actions.tsx

```diff
@@ -195,61 +194,59 @@ export function AlertActions({
           defaultMessage: 'More actions',
         });
 
+  const actionsButtonForPopover = (
+    <EuiToolTip content={actionsToolTip}>
+      <EuiButtonIcon
+        data-grid-tab-managed={true}
+        aria-label={actionsToolTip}
+        color="text"
+        data-test-subj="alertsTableRowActionMore"
+        display="empty"
+        iconType="boxesHorizontal"
+        onClick={() => toggleActionsPopover}
+        size="s"
+      />
+    </EuiToolTip>
+  );
+
   return (
     <>
       {viewInAppUrl && !isInApp ? (
-        <EuiFlexItem>
-          <EuiToolTip
-            content={i18n.translate('xpack.observability.alertsTable.viewInAppTextLabel', {
+        <EuiToolTip
+          content={i18n.translate('xpack.observability.alertsTable.viewInAppTextLabel', {
+            defaultMessage: 'View in app',
+          })}
+        >
+          <EuiButtonIcon
+            iconType="eye"
+            data-test-subj="o11yAlertActionsButton"
+            data-euigrid-tab-managed={true}
+            onMouseOver={handleViewInAppUrl}
+            onClick={() => handleViewInAppUrl()}
+            aria-label={i18n.translate('xpack.observability.alertsTable.viewInAppTextLabel', {
               defaultMessage: 'View in app',
             })}
-          >
-            <EuiButtonIcon
-              data-test-subj="o11yAlertActionsButton"
-              aria-label={i18n.translate('xpack.observability.alertsTable.viewInAppTextLabel', {
-                defaultMessage: 'View in app',
-              })}
-              color="text"
-              onMouseOver={handleViewInAppUrl}
-              href={viewInAppUrl}
-              iconType="eye"
-              size="s"
-            />
-          </EuiToolTip>
-        </EuiFlexItem>
+            color="text"
+            href={viewInAppUrl}
+            size="s"
+          />
+        </EuiToolTip>
       ) : null}
 
-      <EuiFlexItem
-        css={{
-          textAlign: 'center',
-        }}
+      <EuiPopover
+        anchorPosition="downLeft"
+        button={actionsButtonForPopover}
+        closePopover={closeActionsPopover}
+        isOpen={isPopoverOpen}
+        panelPaddingSize="none"
+        onClick={() => setIsPopoverOpen(!isPopoverOpen)}
```

Comment:
What do you think about changing this to use the helper function like you did on L#206?

```suggestion
        onClick={() => toggleActionsPopover}
```